nixos/modules/services/networking/wireguard.nix

```diff
@@ -223,7 +223,30 @@ let
         type = with types; nullOr str;
         description = lib.mdDoc ''
           Endpoint IP or hostname of the peer, followed by a colon,
-          and then a port number of the peer.
+          and then a port number of the peer. Mutually excludive with
+          "endpointFile".
+
+          Warning for endpoints with changing IPs:
+          The WireGuard kernel side cannot perform DNS resolution.
+          Thus DNS resolution is done once by the `wg` userspace
+          utility, when setting up WireGuard. Consequently, if the IP address
+          behind the name changes, WireGuard will not notice.
+          This is especially common for dynamic-DNS setups, but also applies to
+          any other DNS-based setup.
+          If you do not use IP endpoints, you likely want to set
+          {option}`networking.wireguard.dynamicEndpointRefreshSeconds`
+          to refresh the IPs periodically.
+        '';
+      };
+
+      endpointFile = mkOption {
+        default = null;
+        example = "/run/secrets/peer.address";
+        type = with types; nullOr str;
+        description = lib.mdDoc ''
+          File Containing the endpoint IP or hostname of the peer,
```

Comment:
question: does this support IPv6? say I put this in the `endpointFile`:

```
[2001:db8::1]:23542
```

would that work?

Reply:
I made a local change to the `file-config` test to make it use IPv6, and it worked.  In general, this should support anything that's usable by directly specifying it in `endpoint`.